src/transport/RendezvousSession.cpp

```diff
@@ -386,21 +391,18 @@ CHIP_ERROR RendezvousSession::WaitForPairing(Optional<NodeId> nodeId, uint32_t s
 CHIP_ERROR RendezvousSession::Pair(Optional<NodeId> nodeId, uint32_t setupPINCode)
 {
     UpdateState(State::kSecurePairing);
-    ReleasePairingSessionHandle();
     return mPairingSession.Pair(mParams.GetPeerAddress(), setupPINCode, nodeId, mParams.GetRemoteNodeId().ValueOr(kUndefinedNodeId),
                                 mNextKeyId++, this);
 }
 
 void RendezvousSession::SendNetworkCredentials(const char * ssid, const char * passwd)
 {
     mNetworkProvision.SendNetworkCredentials(ssid, passwd);
-    ReleasePairingSessionHandle();
```

Comment:
Why is removing this line, and the one in SendThreadCredentials, ok?

Reply:
Talked to @bzbarsky-apple offline. To close the loop here: this was precautionary check, but reviewing the code further revealed it was not needed